### PR TITLE
Legend Patch Shapes in Style Manager

### LINKED ISSUE #5329: [FEATURE] Support legend patch shapes in style manager databases
Original commit: https://github.com/qgis/QGIS/commit/8a0e007ba5d14edc07c989b28d2351ec575e2132 by nyalldawson

And clean up a lot of duplicate code which makes adding new types
to QgsStyle painful

### PR DESCRIPTION
**Pull request by felixfeckler**
fixes #5329

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is requested

### CODE REVIEW

**Comment by felixfeckler** [comment]
Thought about adding more infos into: [Customizing legend items](https://docs.qgis.org/testing/en/docs/user_manual/print_composer/composer_items/composer_legend.html#customizing-legend-items )
But actual the required infos are present?

**Comment by roya0045** [comment]
@felixfeckler unrelated to this PR but you're doing a great Job consolidating the doc ( that also Goes for you Harrissou, as usual ;) )

**Comment by DelazJ** [comment]
> Thought about adding more infos into: [Customizing legend items](https://docs.qgis.org/testing/en/docs/user_manual/print_composer/composer_items/composer_legend.html#customizing-legend-items)
But actual the required infos are present?

Hmmm not really sure. To me, the style manager is the place to describe what and how the item can be configured/designed (I recall a user wondering what these numbers in the widget meant or are supposed to be edited); the text in layout legend item section is simply an "application area" and afaict, this is what its current description does.
IOW I think we should expand on this feature in the stye manager, with preferably a subsection (to make it kind-of pair with other tabs in the style manager, and have a place we can connect a help button from the dialog to)

**Comment by felixfeckler** [comment]
Hmm not sure why the build failed... on local it builds without error messages...

File: docs/user_manual/style_library/style_manager.rst [hunk 1]
```diff
@@ -324,6 +327,55 @@ choose the ramp type:
  stop` as well as :kbd:`DEL` key removes the selected color stop. 
 
 
+.. _legend_patch:
+
+Creating a Legend Patch Shape
+=============================
+
+To create a new Legend Patch Shape, activate the Legend Patch Shapes tab and
+click the |signPlus| :sup:`Add item` button. The button reveals a drop-down
```
Comment (DelazJ) [inline docs/user_manual/style_library/style_manager.rst:336]: signPlus should be symbologyAdd, with a reference at the bottom, hence the failing build

**Pull request merged**